backend/internal/controllers/user.go

```diff
@@ -65,3 +67,95 @@ func (c *UserController) CreateUser(rw http.ResponseWriter, req *http.Request) {
 
 	writeResponse(rw, http.StatusCreated, newUser)
 }
+
+/*
+ * Handles GET requests, requires {id}
+ * Sends JSON object with all user data if request is authenticated
+ */
+func (c *UserController) GetUser(rw http.ResponseWriter, req *http.Request) {
+	var userID string = chi.URLParam(req, "id")
+
+	// Check for valid token
+	uid, ok := req.Context().Value("user").(string)
+
+	if !ok {
+		writeResponse(rw, http.StatusInternalServerError, "error")
+		fmt.Println("unable to get user from request context")
+		return
+	}
+
+	if uid != userID {
+		writeResponse(rw, http.StatusForbidden, "Token does not match request")
+		return
+	}
+
+	// Fetch user info
+	user, err := c.Users.FetchUserByID(req.Context(), userID)
+	if err != nil {
+		writeResponse(rw, http.StatusInternalServerError, "error")
+		fmt.Println(err)
+		return
+	}
+
+	// User not found
+	if user == nil {
+		writeResponse(rw, http.StatusNotFound, "user does not exist")
+		return
+	}
+
+	writeResponse(rw, http.StatusOK, user)
+}
+
+/*
+ * Handles PATCH request, requires {id}
+ * Update fields of user info
+ */
+func (c *UserController) UpdateUser(rw http.ResponseWriter, req *http.Request) {
+	var userID string = chi.URLParam(req, "id")
+
+	var user models.User
```

Comment:
Hmm, I think we should define a new struct for this route. Re-using models.User here has two consequences:

1. It allows the request to specify a new ID or email, which isn't allowed, so you have to prevent this manually as you've done before
2. It doesn't allow you to specify null values, making partial updates (which is something PATCH should allow) impossible.

Instead, let's make an UpdateUser struct that includes only the fields you're allowed to update, and gives each field a pointer type (so they can be nil in partial updates).